stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -563,16 +608,127 @@ impl NakamotoBlock {
+                "Invalid block -- {} tenure txs",
+                tenure_change_positions.len()
+            );
+            return Some(false);
+        }
+
+        let tc_idx = 0;
+        if tenure_change_positions != vec![tc_idx] {
+            // invalid -- wrong placement
+            debug!(
+                "Invalid block -- tenure txs at {:?}, expected {:?}",
+                &tenure_change_positions,
+                &vec![tc_idx]
+            );
+            return Some(false);
+        }
+
+        let TransactionPayload::TenureChange(tc_payload) = &self.txs[tc_idx].payload else {
+            // this transaction is not a tenure change
+            // (should be unreachable)
+            debug!(
+                "Invalid block -- tx at index {} is not a tenure tx",
+                &tc_idx
+            );
+            return Some(false);
+        };
+        if tc_payload.cause != TenureChangeCause::Extended {
+            // not a tenure-extend, and can't be valid since all other tenure-change types require
+            // a coinbase (which is not present)
+            debug!("Invalid block -- tenure tx is not a tenure extension");
+            return Some(false);
+        }
+
+        if tc_payload.previous_tenure_end != self.header.parent_block_id {
+            // discontinuous
+            debug!(
+                "Invalid block -- discontiguous: {} != {}",
+                &tc_payload.previous_tenure_end, &self.header.parent_block_id
+            );
+            return Some(false);
+        }
+
+        if tc_payload.tenure_consensus_hash != self.header.consensus_hash
+            || tc_payload.prev_tenure_consensus_hash != self.header.consensus_hash
+        {
+            // tenure-extends don't change the current miner
+            debug!(
+                "Invalid block -- expected {} = {} && {} = {}",
+                &tc_payload.tenure_consensus_hash,
+                &self.header.consensus_hash,
+                &tc_payload.prev_tenure_consensus_hash,
+                &self.header.consensus_hash
+            );
+            return Some(false);
+        }
+
+        Some(true)
+    }
+
     /// Determine if this is a well-formed first block in a tenure.
-    /// * It has one or more TenureChange transactions
-    /// * It then has a coinbase
+    /// * It has exactly one TenureChange, and it requires a sortition and points to the parent of
+    /// this block (this checks `cause` and `previous_tenure_end`)
+    /// * It then has a Nakamoto coinbase
     /// * Coinbases and TenureChanges do not occur anywhere else
     ///
     /// Returns Some(true) if the above are true
     /// Returns Some(false) if this block has at least one coinbase or TenureChange tx, but one of
     /// the above checks are false
     /// Returns None if this block has no coinbase or TenureChange txs
-    pub fn is_wellformed_first_tenure_block(&self) -> Option<bool> {
+    pub fn is_wellformed_tenure_start_block(&self) -> Option<bool> {
```

Comment:
As in the other `is_wellformed_` method, anything returning `Some(false)` needs to emit a `warn!` (rather than `debug!`) and should be descriptive enough that someone can understand why the block was invalid without looking at this code.